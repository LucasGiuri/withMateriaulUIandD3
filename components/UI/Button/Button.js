import PropTypes from 'prop-types';
import { Btn } from './Button.styles';

const Button = ({text, isDisabled, onClick}) => {
  const onClickButton = () => {
    if (!isDisabled) {
      onClick();
    }
  };

  return (
    <Btn onClick={onClickButton} isDisabled={isDisabled}>{text}</Btn>
  )
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func
};

Button.defaultProps = {
  isDisabled: false,
  onClick: () => {}
};

export default Button;