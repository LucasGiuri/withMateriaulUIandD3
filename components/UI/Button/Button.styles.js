import styled from 'styled-components';
import { WHITE, BLACK, OLD_GREY } from '../../../public/styles/colors';
import { device } from '../../../public/styles/devices';

export const Btn = styled.button`
  padding: 10px;
  color: ${props => props.isDisabled ? OLD_GREY : WHITE};
  cursor: ${props => props.isDisabled ? 'default' : 'pointer'};
  border-color: ${props => props.isDisabled ? OLD_GREY : WHITE};
  background-color: ${BLACK};
  font-size: 14px;
  
  @media ${device.mobile} {
    width: 49%;
  }
  
  @media ${device.tablet} {
    width: auto;
    margin-left: 14px;
  }
`;