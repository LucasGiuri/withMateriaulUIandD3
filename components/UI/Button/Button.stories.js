import React from 'react';
import { storiesOf } from '@storybook/react';
import Button from './Button';

const text = "Button";

storiesOf('Button')
  .add('Button default', () => (
    <Button text={text} />
  ))
  .add('Button disabled', () => (
    <Button text={text} isDisabled={true} />
  ))
