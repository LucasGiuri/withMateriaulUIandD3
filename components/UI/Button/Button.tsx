 import { Btn } from './Button.styles';

interface ButtonProps {
  text: string;
  isDisabled: boolean;
  onClick: () => void;
};

function Button(props: ButtonProps) {
  const { text, isDisabled, onClick } = props;

  function onClickButton(): void {
    if (!isDisabled) {
      onClick();
    }
  };

  return (
    <Btn onClick={onClickButton} isDisabled={isDisabled}>{text}</Btn>
  )
};

export default Button;