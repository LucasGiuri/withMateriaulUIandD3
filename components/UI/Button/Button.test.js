import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';
import 'jest-styled-components';

describe('Button', () => {
  const text = "Button";
  const isDisabled = true;

  it('Button basic render snapshoot', () => {
    const component = renderer.create(<Button text={text} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Button isDisabled render snapshoot', () => {
    const component = renderer.create(<Button text={text} isDisabled={isDisabled} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Button onClick is clicked', () => {
    const onClick = jest.fn();
    const component = (<Button text={text} onClick={onClick} />);
    const tree = renderer.create(component);
    const button = tree.root.findByType(Button);
    button.props.onClick();
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith();
  })
})
