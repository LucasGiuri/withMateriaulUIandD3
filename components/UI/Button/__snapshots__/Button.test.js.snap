// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button basic render snapshoot 1`] = `
.c0 {
  padding: 10px;
  color: #FFFFFF;
  cursor: pointer;
  border-color: #FFFFFF;
  background-color: #000000;
  font-size: 14px;
}

@media (min-width:320px) {
  .c0 {
    width: 49%;
  }
}

@media (min-width:768px) {
  .c0 {
    width: auto;
    margin-left: 14px;
  }
}

<button
  className="c0"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button Button isDisabled render snapshoot 1`] = `
.c0 {
  padding: 10px;
  color: #999999;
  cursor: default;
  border-color: #999999;
  background-color: #000000;
  font-size: 14px;
}

@media (min-width:320px) {
  .c0 {
    width: 49%;
  }
}

@media (min-width:768px) {
  .c0 {
    width: auto;
    margin-left: 14px;
  }
}

<button
  className="c0"
  onClick={[Function]}
>
  Button
</button>
`;
